### ISSUE TITLE
Examples on moving large structured data between Java/Obj-C and Dart

### ISSUE THREAD

**Issue by eseidelGoogle** [flutter/flutter#18850]
This comes up when authoring plugins, or using existing 3rd party frameworks.  Sometimes one wants to move large amounts of structured data in/out of Dart.

I've heard at least anecdotal reports that this is slow (I've not personally benchmarked this).

I believe lack of documentation/direction here is also part of what is motivating upvotes for https://github.com/flutter/flutter/issues/7053.

No urgency on this, just a percieved opportunity.  FYI @RedBrogdon @mjohnsullivan

**Comment by eseidelGoogle** [comment]
Two possible options to recommend could be flatbuffers (thanks @dnfield!) or protobuffs (which I believe some Google-internal apps use for plugins, @mehmetf would know).

**Comment by mehmetf** [comment]
Yes, protobufs are used widely internally. Links off of https://pub.dartlang.org/packages/protobuf explain how to use them outside Google.

**Comment by zoechi** [comment]
Could #18325 be related?

**Comment by eseidelGoogle** [comment]
I don't care how it's implemented.  But I'd look for an example of how to move some big pile of data from Obj-C or Java into Dart.  Maybe just pointing to the camera plugin?  Or something which moves a blob from heap to heap w/o going through a file (I've heard separate complaints about needing to go through a file to move large data -- which I believe is what the camera plugin currently does?)
